packages/forms/src/documents/pdf/parsing-api.ts

```diff
@@ -122,66 +152,82 @@ export const callExternalParser = async (
     patterns: {},
     outputs: {},
     root: 'root',
-    title: extracted.title,
+    title: extracted.form_summary.title,
   };
 
   const rootSequence: PatternId[] = [];
+
   for (const element of extracted.elements) {
+    const random_id = Math.random().toString(36).substring(2, 18); // TODO: verify we like this
```

Comment:
Using a random id for elements that don't have a pdf field id. We could do this differently, but I like seeing at a glance which ids are pdf assembly related and which aren't.

If we prefer, I can just wrap this into the api and we can treat all ids as canonical from this side.

Reply:
I've been using `crypto.randomUUID()` for that purpose so far - there's a function (which isn't exported), called `generatePatternId` in `pattern.ts` that could be used. I think it makes sense for the API to only provide the IDs as specified in the PDF, so doing this here should be fine for now.